script/DeployMaxVault.s.sol

```diff
@@ -148,7 +147,17 @@ contract DeployMaxVault is BaseScript {
             setWithdrawRule(vault, contracts.YNCLISBNBK());
             setDepositAssetRule(vault, contracts.YNCLISBNBK(), contracts.WBNB());
             setWithdrawAssetRule(vault, contracts.YNCLISBNBK(), contracts.WBNB());
-            setApprovalRule(vault, contracts.YNCLISBNBK(), contracts.WBNB());
+        }
+
+        if (contracts.YNWBNBK() != address(0x0b) && contracts.YNCLISBNBK() != address(0x0c)) {
+            address[] memory underlyingVaults = new address[](2);
+            underlyingVaults[0] = contracts.YNWBNBK();
+            underlyingVaults[1] = contracts.YNCLISBNBK();
+            setApprovalRule(vault, contracts.WBNB(), underlyingVaults);
```

Comment:
instead of elseif elseif 

Once you get past the revert you have only one if that sets to 2 items or just one the underlyingVaults array



after if, call underlyingVaults